mesonbuild/interpreter.py

```diff
@@ -2805,16 +2805,16 @@ def _find_cached_fallback_dep(self, name, dirname, varname, wanted, required):
     def func_dependency(self, node, args, kwargs):
         self.validate_arguments(args, 1, [str])
         name = args[0]
+        display_name = name if name else '(anonymous)'
```

Comment:
Instead of giving it a name for display, maybe we should not print messages for this at all, and only log to the debug log? Users probably will not understand what `(anonymous)` means.

Reply:
I'm not sure being silent would be a good idea.  As you pointed out, the fallback subproject could be optional, so we need to give some indication of what we are doing.

But yeah, something clearer than "anonymous" would be better, perhaps something like "unnamed dependency"